File: concurrent/src/main/java/com/mars/concurrency/first/chapter10/Lock.java
package com.mars.concurrency.first.chapter10;

import java.util.Collection;

/***************************************
 * @author:Alex Wang
 * @Date:2017/2/22 QQ:532500648
 * QQ交流群:286081824
 ***************************************/
public interface Lock {

    void lock() throws InterruptedException;

    void lock(long mills) throws InterruptedException, TimeOutException;

    void unlock();

    Collection<Thread> getBlockedThread();

    int getBlockedSize();

    class TimeOutException extends Exception {

        public TimeOutException(String message) {
            super(message);
        }
    }

}

File: concurrent/src/test/java/com/mars/concurrency/book17/Lock.scala
package com.mars.concurrency.book17

/**
  * Created by ssk on 2019/1/17 0017.
  * desc:
  */
trait Lock {
  /**
    * 获取显示锁，没有获取锁的线程将被阻塞
    *
    * @throws java.lang.InterruptedException
    */
  @throws(classOf[InterruptedException])
  def lock()

  /**
    * 释放获取的锁
    */
  def unlock()
}
